pkg/internal/apis/certmanager/validation/certificate.go

```diff
@@ -165,6 +176,14 @@ func validateUsages(a *internalcmapi.CertificateSpec, fldPath *field.Path) field
 	return el
 }
 
+func validateSecretTemplateLabels(crt *internalcmapi.CertificateSpec, fldPath *field.Path) field.ErrorList {
+	return metavalidation.ValidateLabels(crt.SecretTemplate.Labels, fldPath.Child("secretTemplate", "labels"))
+}
+
+func validateSecretTemplateAnnotations(crt *internalcmapi.CertificateSpec, fldPath *field.Path) field.ErrorList {
+	return apivalidation.ValidateAnnotations(crt.SecretTemplate.Annotations, fldPath.Child("secretTemplate", "annotations"))
+}
```

Comment:
Thanks for adding this @jonathansp 

I'm just testing it now.

Will you also add some unit tests for these new validations to `certificate_test.go` ?

Do you agree that we should dissallow the setting of `cert-manager.io/` annotations in the secret template?

If so, please add that to the annotation validation.

```
    cert-manager.io/alt-names: example.com
    cert-manager.io/certificate-name: selfsigned-cert
    cert-manager.io/common-name: ""
    cert-manager.io/ip-sans: ""
    cert-manager.io/issuer-group: ""
    cert-manager.io/issuer-kind: Issuer
    cert-manager.io/issuer-name: test-selfsigned
    cert-manager.io/uri-sans: ""

```

More comments to follow.

Reply:
Thanks for reviewing @wallrj 
Just to clarify, any annotation _cert-manager.io/*_ should not be allowed, right? Do you see any restrictions regarding the labels?

Yes, I will add some tests to certificate_test later today.

Reply:
> Just to clarify, any annotation _cert-manager.io/*_ should not be allowed, right? 

Yes, I think we should dissallow any annotation key with prefix `cert-manager.io/` to avoid people accidentally / deliberately interfering with the annotations that cert-manager uses. 

> Do you see any restrictions regarding the labels?

I can't think of any, but I'll check.

> Yes, I will add some tests to certificate_test later today.

:+1:  Thanks.

Reply:
Tests added 👍